### PR TITLE
EVG-16907: handle container secrets when creating and copying projects

### PR DESCRIPTION
**Pull request by Kimchelly**
Jira: https://jira.mongodb.org/browse/EVG-16907

### Description 
* When creating a new project (either a copy or a fresh one), give it a new pod secret for authenticating pods running that project's tasks.
* When copying a project, ensure the new project has a copy of the old project's repo creds.
* Add validation to ensure a project can't end up with multiple pod secrets.
* Remove container secret support from the legacy UI route since container secrets won't be added to the legacy UI.
* Modify some pod utilities in the cloud package to make it easier to ensure that tests use the fake ECS/Secrets Manager services. It's not always that easy to directly pass in the fake clients everywhere (e.g. we can't pass the mock Secrets Manager client into the GraphQL resolvers). To make unit testing easier, I made it possible to choose the client implementation based on the admin settings. By default, tests that use a `mock.Environment` or `testutil.NewEnvironment` will use clients connected to the fake AWS services. A real app server will never set this value, so the actual apps will use real AWS services.

### Testing
* Added unit tests.

### CODE REVIEW

File: rest/data/project.go [hunk 1]
```diff
@@ -75,6 +80,39 @@ func CreateProject(ctx context.Context, projectRef *model.ProjectRef, u *user.DB
 			return err
 		}
 	}
+
+	cachedSettings := env.Settings()
```
Comment (hadjri) [inline rest/data/project.go:84]: Curious why we are caching the value rather than passing a `env.Settings()` into each function?

**Review by hadjri — COMMENTED** [review]
Couple small comments. LGTM otherwise

File: model/project_ref.go [hunk 2]
```diff
@@ -3006,19 +3011,26 @@ func ValidateContainerSecrets(settings *evergreen.Settings, projectID string, or
...
-		}
+		extName, err := newContainerSecretExternalName(settings.Providers.AWS.Pod.SecretsManager, projectID, updatedSecret)
+		catcher.Add(err)
+		updatedSecret.ExternalName = extName
 		updatedSecret.ExternalID = ""
 
 		combined = append(combined, updatedSecret)
 	}
 
+	catcher.NewWhen(numPodSecrets > 1, "a project can have at most one pod secret")
```
[the remark this answers is not in the document]
Comment (Kimchelly) [reply model/project_ref.go]: Done.

File: rest/data/project.go (hunk 1 above)
Comment (Kimchelly) [reply rest/data/project.go:84]: The name is to distinguish from the admin settings that are loaded on L58 using `evergreen.GetConfig()`, which returns the settings directly from the DB, as opposed to `env.Settings()`, which returns the settings as they existed when the app server first started.

File: rest/data/project_settings.go [hunk 3]
```diff
@@ -351,6 +353,62 @@ func DeleteContainerSecrets(ctx context.Context, v cocoa.Vault, pRef *model.Proj
...
+		return nil, errors.Wrap(err, "validating new container secrets")
+	}
+
+	return validated, nil
+}
+
+// newPodSecret returns a new default pod secret with a random value to be
+// stored.
+func newPodSecret(settings *evergreen.Settings, projectID string) model.ContainerSecret {
+	return model.ContainerSecret{
```
[the remark this answers is not in the document]
Comment (Kimchelly) [reply rest/data/project_settings.go:405]: Good catch, fixed.

File: service/project.go [hunk 4]
```diff
@@ -665,7 +613,6 @@ func (uis *UIServer) modifyProject(w http.ResponseWriter, r *http.Request) {
 	projectRef.PeriodicBuilds = []model.PeriodicBuildDefinition{}
 	projectRef.PerfEnabled = &responseRef.PerfEnabled
 	projectRef.ContainerSizes = containerSizes
-	projectRef.ContainerSecrets = allContainerSecrets
```
[the remark this answers is not in the document]
Comment (Kimchelly) [reply service/project.go:668]: Yeah, since we're only supporting the new UI, I removed the container size logic.

**Review by hadjri — APPROVED** [review]
LGTM with merge conflict resolution

**Pull request merged**